File: ts/project/.sdk/tm/py/utility/prepare_headers.py
# ProjectName SDK utility: prepare_headers

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs


def prepare_headers_util(ctx):
    options = ctx.client.options_map()
    headers = vs.getprop(options, "headers")

    if headers is None:
        return {}

    out = vs.clone(headers)
    if isinstance(out, dict):
        return out
    return {}

File: ts/project/.sdk/tm/csharp/utility/PrepareHeaders.cs
// ProjectName SDK utility: prepareHeaders.

using Voxgig.Struct;

namespace ProjectNameSdk.Util;

public static partial class SdkUtility
{
    internal static Dictionary<string, object?> PrepareHeadersUtil(Context ctx)
    {
        var options = ctx.Client!.OptionsMap();

        var headers = StructUtils.GetProp(options, "headers");
        if (headers == null)
        {
            return new Dictionary<string, object?>();
        }

        return StructUtils.Clone(headers) as Dictionary<string, object?>
            ?? new Dictionary<string, object?>();
    }
}
